app/assets/javascripts/components/forms/campaign_page_form.js.jsx

```diff
@@ -27,19 +21,52 @@ var EditCampaignPage = React.createClass({
     return { flux: flux };
   },
 
+  componentDidMount(){
+    this.onChangeTimeout = null;
+  },
+
+  componentWillUpdate(a, b){
+  },
+
   handleSubmits(data) {
     this.getFlux().actions.updateCampaignPage(data);
   },
 
+  handleSumit(e) {
+    e.preventDefault()
+
+    console.log(this.state.title);
+  },
+
+  saveTitle() {
+    let data = {title: this.state.title };
+    this.getFlux().actions.updateCampaignPage(data);
+  },
+
+  handleChange: function(event) {
+    this.setState({title: event.target.value});
+
+    clearTimeout(this.onChangeTimeout);
+
+    this.onChangeTimeout = setTimeout(function(){
+      this.saveTitle();
+    }.bind(this) , 1000);
+  },
+
   render() {
     return (
-      <div>
-      <form onSubmit={this.handleSubmit}>
-        <CampaignPageForm title={this.props.title} />
-        <CheckBox onChange={this.handleSubmits} name='featured' label='Featured' checked={this.state.featured}/>
-        <CheckBox onChange={this.handleSubmits} name='active' label='Active' checked={this.state.active}/>
-      </form>
-      <div id="new-widget"></div>
+      <div className='campaign-page-config'>
+        <form onSubmit={this.handleSumit}>
+          <div className='page-title'>
+            <input onChange={ this.handleChange } defaultValue={ this.props.title } type="text" id="campaign_page_title" name="campaign_page[title]" ref="title" />
```

Comment:
I feel pretty strongly that updating a field's value on the server every time someone loses focus on a field is an anti-feature. Even if we're updating over AJAX, as a user I much prefer to be able to revise and play around before I commit my work by clicking submit.